src/test/java/com/stripe/functional/UsageRecordTest.java

```diff
@@ -1,65 +1,39 @@
 
-import com.stripe.BaseStripeFunctionalTest;
+import com.stripe.BaseStripeTest;
 import com.stripe.exception.StripeException;
-import com.stripe.model.Customer;
-import com.stripe.model.Plan;
-import com.stripe.model.Subscription;
-import com.stripe.model.SubscriptionItem;
 import com.stripe.model.UsageRecord;
-import java.util.ArrayList;
+import com.stripe.net.APIResource;
+
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import org.junit.Test;
-
-public class UsageRecordTest extends BaseStripeFunctionalTest {
-  @Test
-  public void testUsageRecordCreate() throws StripeException {
-    Plan plan = createMeteredPlan();
-    Subscription sub = createSubscription(plan);
-    SubscriptionItem subItem = sub.getSubscriptionItems().getData().get(0);
-
-    Map<String, Object> params = new HashMap<>();
-    params.put("quantity", 1000);
-    long unixTime = System.currentTimeMillis() / 1000L;
-    params.put("timestamp", unixTime);
-    params.put("subscription_item", subItem.getId());
-
-    UsageRecord ur = UsageRecord.create(params, null);
-
-    assertEquals(new Long(1000), ur.getQuantity());
-    assertEquals(new Long(unixTime), ur.getTimestamp());
-  }
 
-  private Plan createMeteredPlan() throws StripeException {
-    Map<String, Object> productParams = new HashMap<String, Object>();
-    productParams.put("name", "Bar");
-
-    Map<String, Object> params = getUniquePlanParams();
-    params.remove("name");
-    params.put("nickname", "Foo");
-    params.put("product", productParams);
-
-    params.put("usage_type", "metered");
-
-    return Plan.create(params);
-  }
-
-  private Subscription createSubscription(Plan plan) throws StripeException {
-    final Customer customer = Customer.create(defaultCustomerParams);
-    Map<String, Object> subscriptionParams = new HashMap<String, Object>();
-
-    Map<String, Object> planEntry = new HashMap<>();
-    planEntry.put("plan", plan.getId());
+import org.junit.Test;
 
-    List<Map<String, Object>> subItems = new ArrayList<>();
-    subItems.add(planEntry);
+public class UsageRecordTest extends BaseStripeTest {
+  private static final String SUBCRIPTION_ITEM_ID = "si_123";
 
-    subscriptionParams.put("items", subItems);
-    subscriptionParams.put("customer", customer.getId());
-    return Subscription.create(subscriptionParams);
+  @Test
+  public void testCreate() throws StripeException {
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("quantity", 10);
+    params.put("subscription_item", SUBCRIPTION_ITEM_ID);
+    params.put("timestamp", System.currentTimeMillis() / 1000L);
+
+    UsageRecord resource = UsageRecord.create(params, null);
+
+    // UsageRecord.create() has a non-standard behavior: it uses the `subscription_item` element
+    // in the params map to form the endpoint's URL. We need to remove it from the map because
+    // it won't be sent as a URL-encoded parameter.
+    params.remove("subscription_item");
```

Comment:
flagging that we should fix this in the next major version and move back to a parameter for the id